Fixes for #1209

tharakadesilva opened the pull request:
Fixes #1209

tkrop commented:
:+1:

vadeg commented on server/zally-core/src/main/kotlin/org/zalando/zally/core/ast/ReverseAst.kt:
```diff
@@ -48,6 +49,9 @@ class ReverseAst internal constructor(
     }
 
     companion object {
+        const val pathsJsonPointerString = "/paths"
```
Please follow [kotlin guidelines](https://kotlinlang.org/docs/coding-conventions.html#property-names) for naming.

tharakadesilva replied:
Fixed.

vadeg commented on server/zally-core/src/main/kotlin/org/zalando/zally/core/ast/ReverseAst.kt:
```diff
@@ -48,6 +49,9 @@ class ReverseAst internal constructor(
 
     companion object {
+        const val pathsJsonPointerString = "/paths"
+        val pathsJsonPointer = pathsJsonPointerString.toJsonPointer()
```
Same here about Kotlin guidelines.

tharakadesilva replied:
Fixed.

vadeg reviewed (changes requested):
Hi @tharakadesilva !

Thank you for your contribution, but I don't think that `ReverseAst` is the right place to define `paths` property. This is a part of OpenAPI / Swagger spec. I think it should be somewhere in the `Context` or as an extension for `OpenAPI` or `Swagger` classes.

tharakadesilva commented:
> Hi @tharakadesilva !
> 
> Thank you for your contribution, but I don't think that `ReverseAst` is the right place to define `paths` property. This is a part of OpenAPI / Swagger spec. I think it should be somewhere in the `Context` or as an extension for `OpenAPI` or `Swagger` classes.

Hi @vadeg, thank you for the feedback. I am a little torn on where to move these. I feel that the recommendation to move it as an extension to `OpenAPI` is the best approach. `OpenAPI` already has some extensions in the `OpenApiUtil.kt` file. Do you think that this might be a good place to place these values as well?

vadeg commented:
@tharakadesilva I would suggest something like this
```
class OpenApiPaths {
    companion object {
        const val PATHS = "/paths"
    }
}
```
Unfortunately, Kotlin [doesn't support](https://youtrack.jetbrains.com/issue/KT-11968) static extensions for Java classes and the original idea to make someting like 
```
io.swagger.v3.oas.models.Paths
val Paths.paths: String 
    get() = "sdafa"
```
and use like `Paths.paths` will not work.

tharakadesilva commented:
@vadeg I have fixed the code as you suggested. Would appreciate it if you could review it again now.

vadeg commented on server/zally-core/src/main/kotlin/org/zalando/zally/core/util/OpenApiPaths.kt:
```diff
@@ -0,0 +1,10 @@
+class OpenApiPaths {
+    companion object {
+        const val PATHS_JSON_POINTER_STRING = "/paths"
+        val PATHS_JSON_POINTER = PATHS_JSON_POINTER_STRING.toJsonPointer()
```
I thinks this is not needed here, because it is up to the client how to interpret the string.

tharakadesilva replied:
I have removed this.

vadeg commented on server/zally-core/src/main/kotlin/org/zalando/zally/core/util/OpenApiPaths.kt:
```diff
@@ -0,0 +1,10 @@
+
+class OpenApiPaths {
+    companion object {
+        const val PATHS_JSON_POINTER_STRING = "/paths"
```
This is not a JSON pointer but a json path. Could you rename it to `PATHS`?

tharakadesilva replied:
@vadeg I renamed this to `PATHS`. However, now I feel that the class name `OpenApiPaths` might not appropriate anymore. I was thinking about renaming `OpenApiSections`, what do you think?

vadeg replied:
Good idea.

tharakadesilva replied:
Fixed.

vadeg reviewed (commented):
Hi @tharakadesilva 

Sorry for the delay. I have left a couple of comments.

tharakadesilva commented:
> Hi @tharakadesilva
> 
> Sorry for the delay. I have left a couple of comments.

Hi @vadeg, thank you for the feedback. I have addressed the comments and I think it is ready for a review again. Thank you for the continuous feedback :)

vadeg reviewed (approved):
@tharakadesilva thank you very much for your contribution!

vadeg commented:
👍

tkrop commented:
:+1:
